Enable dev mode by default

vsavkin opened the pull request:
Closes #5883

IgorMinar commented on modules/angular2/src/core/application_ref.ts:
```diff
@@ -425,7 +426,15 @@ export class ApplicationRef_ extends ApplicationRef {
+      let c = this._injector.get(Console);
+      let modeDescription =
+          assertionsEnabled() ?
+              `in the development mode. Call enableProdMode() to enable the production mode.` :
```
use `"` (double-quoted string) for consistency with the next line

vsavkin replied:
fixed

IgorMinar commented on modules/angular2/src/facade/lang.dart:
```diff
@@ -266,6 +266,16 @@ void enableDevMode() {
 
+void enableProdMode() {
+  if (_forceDevMode) {
+    return;
```
shouldn't we throw? why make it look like the prod mode was turned on when it wasn't?

IgorMinar commented on modules/angular2/src/facade/lang.ts:
```diff
@@ -39,7 +39,7 @@ export function getTypeNameForDebugging(type: Type): string {
 export var Math = _global.Math;
 export var Date = _global.Date;
 
-var _devMode: boolean = !!_global.angularDevMode;
```
this is a breaking change - I believe we rely on this in our doc examples or in the playground apps - I'm not sure. @alxhub?

vsavkin replied:
It is. Enabling dev mode by default is a breaking change. No way around it.

alxhub replied:
Our tests set `global.angularDevMode` to force dev mode. Enabling it by default shouldn't break them because that's what they want anyway.

You should probably remove `angularDevMode` entirely then (`test-main.js`, `cjs-jasmine/index.js` and various broccoli trees), as dev mode by default obviates them.

IgorMinar commented on modules/angular2/src/facade/lang.ts:
```diff
@@ -64,6 +64,21 @@ export function enableDevMode() {
+export function enableProdMode() {
+  if (_devModeLocked) {
+    // Cannot use BaseException as that ends up importing from facade/lang.
+    throw 'Cannot disabled dev mode after platform setup.';
```
why do you throw here but not in the dart version?

IgorMinar replied:
nevermind - wrong `if`. this is correct and consistent with the dart version

IgorMinar commented on modules/angular2/src/facade/lang.ts:
```diff
@@ -39,7 +39,7 @@ export function getTypeNameForDebugging(type: Type): string {
 export var Date = _global.Date;
 
-var _devMode: boolean = !!_global.angularDevMode;
+var _devMode: boolean = true;
```
do we set it to true in dart as well? I don't see the change in this diff, but maybe it's already that way...

IgorMinar commented on modules/angular2/test/public_api_spec.ts:
```diff
@@ -837,6 +837,8 @@ var NG_CORE = [
   'ComponentRef.location',
   'ComponentRef.location=',
+  'Console',
+  'Console.log()',
```
unless I'm missing something, the console is used only by the core bundle, so why do we need to make it public?

If we don't absolutely have to, I'd keep this as an internal API and consider exposing it in the future.

IgorMinar commented:
If dev mode is on by default, then do we still need the enabledDevMode flag? what is it useful for?

IgorMinar commented on modules/angular2/src/core/dev_mode.ts:
```diff
@@ -1 +1 @@
-export {enableDevMode} from 'angular2/src/facade/lang';
\ No newline at end of file
+export {enableProdMode} from 'angular2/src/facade/lang';
```
change the filename to `prod_mode.ts` ?

IgorMinar commented on modules/angular2/src/facade/lang.dart:
```diff
@@ -249,15 +249,16 @@ bool isJsObject(o) {
 
-bool _forceDevMode = false;
-bool _devModeLocked = false;
+bool _forceDevMode = true;
```
I don't think that we can do this without a g3sync troubles. let's keep this `false` and create an issue to change that after beta.

IgorMinar commented:
lgtm

mary-poppins commented:
Merging PR #5888 on behalf of @alexeagle to branch [presubmit-alexeagle-pr-5888](https://github.com/angular/angular/tree/presubmit-alexeagle-pr-5888).

IgorMinar commented:
CI seems to be failing - dart timeout - I suspect that it could be related to #5896

vsavkin committed:
feat(core): enable dev mode by default

BREAKING CHANGE

Before

Previously Angular would run in dev prod mode by default, and you could enable the dev mode by calling enableDevMode.

After

Now, Angular runs in the dev mode by default, and you can enable the prod mode by calling enableProdMode.
